clang/lib/Driver/ToolChains/PICMid.cpp

```diff
@@ -0,0 +1,151 @@
+//===-- PICMid.cpp - Define ToolChain for PICMid -------*- C++ -*-===//
+//
+// Part of LLVM-PIC, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "PICMid.h"
+
+#include "CommonArgs.h"
+
+#include "clang/Driver/Compilation.h"
+#include "clang/Driver/Driver.h"
+#include "clang/Driver/Options.h"
+#include "llvm/Support/Path.h"
+
+using namespace llvm::opt;
+using namespace clang::driver;
+using namespace clang::driver::tools;
+using namespace clang::driver::toolchains;
+
+PICMidToolChain::PICMidToolChain(const Driver &D, const llvm::Triple &Triple,
+         const llvm::opt::ArgList &Args)
+    : ToolChain(D, Triple, Args) {
+  // Look for binaries in both the installation and driver directory.
+  getProgramPaths().push_back(getDriver().getInstalledDir());
+  if (getDriver().getInstalledDir() != getDriver().Dir)
```

Comment:
Wrap "if body" with squirelly brackets `{}` pls.

Reply:
your are right with the merging